scipy/interpolate/interpolate.py

```diff
@@ -486,6 +488,19 @@ def __init__(self, x, y, kind='linear', axis=-1,
                 self.x_bds = self.x_bds[1:] + self.x_bds[:-1]
 
                 self._call = self.__class__._call_nearest
+            elif kind == 'previous':
+                # Side for np.searchsorted and index for clipping
+                self.side = 'left'
+                self.ind = 0
+                # Move x by one floating point value to the left
+                self.x_bds = np.nextafter(self.x, self.x-1)
```

Comment:
Not sure, does this work if diff(self.x) < 1, would it not skip a point?

Reply:
Not sure what you mean. `np.nextafter(x, x-1)` is equal to `np.nextafter(x, x-1000)` or  `np.nextafter(x, x-np.infty)`, the second argument is only to indicate the direction in which it will return the next floating-point value after x.

E.g., `np.nextafter(1, 1-1) = np.nextafter(1, 1-1000) = 0.99999999999999989`, on my machine.